projects/concurrency/cache-computing/cache.go

```diff
@@ -0,0 +1,65 @@
+package cachecomputing
+
+import (
+	"concurrency/cache"
+	"sync"
+)
+
+type Creator[K comparable, V any] func(K) V
+
+type ComputingCache[K comparable, V any] struct {
+	cache                   cache.Cache[K, V]
+	creator                 Creator[K, V]
+	pendingKeys             map[K]chan V
+	numberOfWaitingRoutines map[K]int
```

Comment:
This works, but often having two maps which are expected to always have the same keys in them is something we avoid, preferring to have one map that contains all of the values from both maps in a structured way. That way we group our data together, and avoid the possibility that a bug means we maps get out of sync.

Can you think of how to achieve the same result with just one map?